oeps/oep-0036-microservice-routing-layer.rst

```diff
@@ -0,0 +1,99 @@
+===================================
+OEP-36: Microservice Routing Layer
+===================================
+
+.. list-table::
+
+  * - OEP
+    - :doc:`oep-0036-microservice-routing-layer`
+  * - Title
+    - Microservice Routing Layer
+  * - Last-Modified
+    - 2019-04-25
+  * - Author
+    - Fred Smith <derf@edx.org>
+  * - Arbiter
+    - 
+  * - Status
+    - Provisional
+  * - Type
+    - Architecture
+  * - Created
+    - 2019-04-25
+
+Abstract
+========
+
+Proposes a system with which users can interact with microservice applications
+using a layer 7 routing framework. 
+
+Motivation
+==========
+
+Microservices are currently implemented using layer 4 (DNS) based routing. Each
+microservice occupies a unique hostname, with dedicated ELB or CDN 
+infrastructure to support routing user traffic to the service. This pattern
+is simple to monitor, deploy, and manage, but slow and expensive to build,
+and regularly causes problems with cross service user information sharing. In
+Addition, modifying or splitting a monolithic service into microservices resuls
+in confusing URL changes and costly migrations.
+
+A layer 7 routing layer would allow us to separate the user facing URL or API
+endpoint from the backend implementation. It would allow microservices to exist
+that only manage a small number of endpoints, without requring developers to 
+build out load balancing or CDN infrastructure. 
+
+We are currently using a layer 7 routing framework for prospectus, allowing
```

Comment:
While this is good to point out during internal conversation, prospectus is not an open source / Open edX feature, so I don't know if it makes sense in this document.

Reply:
I think prospectus, or some similar marketing site CMS, should be a component in openedx.  There were a handful presented at Openedx conf.

Reply:
I agree. It came up a lot during the theming discussion during the summit. I think the conclusion was:
> Everyone wants different features in their marketing site. We (edX) will have our own and will expose the features to build your own. If the community wants to get together and build a common marketing site w/ cms, they are welcome to.

Since edX, wouldn't be using it, we would be bad maintainers of that Open edX project.